src/merge.c

```diff
@@ -862,12 +862,18 @@ static int merge_conflict_invoke_driver(
 		(error = git_odb_write(&oid, odb, buf.ptr, buf.size, GIT_OBJECT_BLOB)) < 0)
 		goto done;
 
+	if (buf.size > UINT32_MAX) {
+		git_error_set(GIT_ERROR_MERGE, "merged file contents too large for index");
+		error = -1;
+		goto done;
+	}
```

Comment:
So this means we're not able to solve merge conflicts on 4GB files? Unlikely, but it could very well be that somebody has written a merge conflict driver for binary files that is able to resolve such conflicts. On the other hand, this is not a new restriction and at least we're being explicit now.

Reply:
Right, I'd forgotten how the index handles files > 4GB.  In fact, we should be truncating the file size to a `uint32_t` in the cache.  I updated this to remove this (incorrect) check.

Reply:
Just to be sure, I can't remember how this looked before. Is this already the updated version? Cannot see any "outdated" marker.

Reply:
Right!  Sorry, I must not have pushed up a fixed.  (It's now pushed up.)